utils/check_pack_uniqueness.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+const path = require('path');
+const glob = require('glob');
+const { 
+  readPackFile, 
+  removeCWDPrefix 
+} = require('./helpers');
+
+/**
+ * Finds the path to all top level pack configs
+ * @returns {String[]} An array of the file paths
+ */
+const findMainPackConfigFiles = () => glob.sync(path.resolve(process.cwd(), 'packs/**/config.+(yml|yaml)')); 
+
+/**
+ * Removes whitespace and punctuation from a string
+ * @returns {String} The string with `-` replacing whitespace and punctuation removed
+ */
+const cleanPackName = (str) =>
+  str
+    .trim()
+    .toLowerCase()
+    .replace(/\s+/g, '-')
+    .replace(/-+/, '-')
+    .replace(/[^a-z0-9-]/g, '');
+
+/**
+ * Returns any packs with matching names
+ * @param {Object[]} namesAndPaths an array of objects containing the path and name of a pack
+ * @returns {Object[]} an array of matching values 
+ */
+const getMatchingNames = (namesAndPaths) => {
```

Comment:
while the code is in front of my eyes, im not sure i understand what this function is doing 🙈 
any chance you could walk me through this in a voice chat?

Reply:
Zack gave me a much more simple way to do this, I have updated the function. Still down to walk through it!

Reply:
if you got the time. dont want to hold this up just to account for me 🙈